src/test/java/com/google/api/generator/engine/JavaCodeGeneratorTest.java

```diff
@@ -0,0 +1,886 @@
+            .setName("LibraryServiceStub")
+            .build();
+    JavaWriterVisitor javaWriterVisitor = new JavaWriterVisitor();
+    libraryServiceStubClass.accept(javaWriterVisitor);
+    assertEquals(javaWriterVisitor.write(), EXPECTED_CLASS_STRING);
+  }
+
+  // Private helpers.
+  private CommentStatement createFileHeader() {
+    String fileheader =
+        "Copyright 2020 Gagpic-generator-java\n\n"
+            + "Licensed description and license version 2.0 (the \"License\");\n\n"
+            + "   https://www.foo.bar/licenses/LICENSE-2.0\n\n"
+            + "Software distributed under the License is distributed on an \"AS IS\" BASIS.\n"
+            + "See the License for the specific language governing permissions and\n"
+            + "limitations under the License.";
+    return CommentStatement.withComment(BlockComment.withComment(fileheader));
+  }
+
+  private CommentStatement createOuterClassJavaDocComment() {
+    return CommentStatement.withComment(
+        JavaDocComment.builder()
+            .addComment("Service Description: This is a test comment.")
+            .addComment("")
+            .addSampleCode("LibraryServiceStub libServiceStub = new LibraryServiceStub()")
+            .addOrderedList(
+                Arrays.asList(
+                    "A \"flattened\" method.",
+                    "A \"request object\" method.",
+                    "A \"callable\" method."))
+            .setDeprecated("This is a deprecated message.")
+            .build());
+  }
+
+  private CommentStatement createPreMethodLineComment(String commentString) {
+    return CommentStatement.withComment(LineComment.withComment(commentString));
+  }
+
+  private List<CommentStatement> createPreMethodJavaDocComment() {
+    return Arrays.asList(
+        CommentStatement.withComment(
+            JavaDocComment.builder()
+                .addComment("Add books to Shelf and check if there is a novel,")
+                .addComment("return string message as whether novel books are added to the shelf.")
+                .addComment("")
+                .addParam("shelf", "The Shelf object to which books will put.")
+                .addParam("stack", "The Stack of the BookKinds.")
+                .build()));
+  }
+
+  private static VaporReference createVaporReference(String pkgName, String name) {
+    return VaporReference.builder().setPakkage(pkgName).setName(name).build();
+  }
+
+  private static Variable createVarFromConcreteRef(ConcreteReference ref, String name) {
+    return Variable.builder().setName(name).setType(TypeNode.withReference(ref)).build();
+  }
+
+  private static Variable createVarFromVaporRef(VaporReference ref, String name) {
+    return Variable.builder().setName(name).setType(TypeNode.withReference(ref)).build();
+  }
+
+  private static Variable createVarFromType(TypeNode ref, String name) {
+    return Variable.builder().setName(name).setType(ref).build();
+  }
+
```

Comment:
Thesehelpers are static because they are used to create the shared static variables defined outside of validJavaClass.